addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -118,13 +118,13 @@ def _compute_website_order_line(self):
             for lines in grouped_order_lines.values():
                 if lines.reward_id.reward_type != 'discount':
                     continue
-                if self.env.user.has_group('sale.group_show_price_subtotal'):
-                    price_unit = sum(lines.mapped('price_subtotal'))
-                else:
-                    price_unit = sum(lines.mapped('price_total'))
                 new_lines += self.env['sale.order.line'].new({
                     'product_id': lines[0].product_id.id,
-                    'price_unit': price_unit,
+                    'tax_id': False,
```

Comment:
Still gotta investigate more (and confirm with PO) whether the taxes should be set or not, but I believe this approach of keeping the amount(s) is better, disabling the fields computation and ensuring the values are the right ones.